lib/route.method.js

```diff
@@ -41,7 +41,8 @@ function createRouteInterceptor(routeMethod) {
          * This function is meant to be invoked in promise chain, which means that any exception thrown by callRouteHandler will not get out of routeInterceptor
          * (will be swallowed), so we need to close the connection with proper error sent to client.
          */
-        function safelyCallRouteHandler() {
+        function safelyCallRouteHandler(authResult) {
+            if (authResult instanceof JSONAPI_Error) return sendError(request, response, authResult);
```

Comment:
I see you're sort of short-circuiting things here, which makes a ton of sense. Why not also do it for Error as well?